### PR TITLE
Updates ghc version to `9.0.2` and stackage version to `lts-19.6`.

### PR DESCRIPTION
**Pull request by rimbi**
## Purpose

Updates ghc version to `9.0.2` and stackage version to `lts-19.6`.

## Changes

- Upgrades the stackage version
- Fixes deprecation warnings.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: src/Network/HTTP2/Client.hs [hunk 1]
```diff
@@ -282,7 +282,7 @@ type PushPromiseHandler =
 -- @ _ <- (withHttp2Stream myClient $ \stream -> StreamDefinition _ _) @
 --
 -- Please refer to 'StreamStarter' and 'StreamDefinition' for more.
-withHttp2Stream :: Http2Client -> StreamStarter a
+withHttp2Stream :: Http2Client -> forall a. StreamStarter a
```
Comment (abizjak) [inline src/Network/HTTP2/Client.hs:285]:
```suggestion
withHttp2Stream :: forall a . Http2Client -> StreamStarter a
```
it is more common to qualify type variables at the outside. Is there a reason not to do that here?

File: stack-lts-19.6.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+resolver: lts-19.6
+packages:
+- '.'
+extra-deps: []
+flags: {}
+extra-package-dbs: []
+extra-include-dirs:
+    - /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include/ffi/
```
Comment (abizjak) [inline stack-lts-19.6.yaml]: This should not be part of the repository.

What problem are you having that this solves? Perhaps there is a better solution. We never needed this before.

Comment (rimbi) [reply stack-lts-19.6.yaml]: It complains about the missing `ffi` includes.

File: src/Network/HTTP2/Client.hs (hunk 1 above)
Comment (rimbi) [reply src/Network/HTTP2/Client.hs:285]: This did not work, unfortunately:

image

File: stack-lts-19.6.yaml (hunk 2 above)
Comment (abizjak) [reply stack-lts-19.6.yaml]: I'm sorry, but I don't think we can have this in the configuration file. It points to an arbitrary place that makes sense on your machine, but is not part of the repository, nor is it relevant for linux and windows platforms.

Can you point to what exactly the error is? We can discuss this in a meeting if it is more convenient.

Comment (rimbi) [reply stack-lts-19.6.yaml]: I see your point. On the other hand, we need a solution for the mac targets. You can just ignore them for now. I'll dive into this problem later.

Comment (abizjak) [reply stack-lts-19.6.yaml]: The client builds fine on an intel mac.

So it is not a general mac issue, it is a specific issue either to M1 macs, or to your setup.

I am not sure what you mean by "just ignore them for now". This PR cannot be merged with this line in it. Do you mean you will remove it?

Comment (rimbi) [reply stack-lts-19.6.yaml]: I mean I'll investigate it more and then we'll discuss it. Till then we'll not merge any of the PRs having this issue.

Comment (rimbi) [reply stack-lts-19.6.yaml]: removed

**Review by abizjak — APPROVED** [review]

**Pull request merged**